Python/ceval.c

```diff
@@ -762,16 +762,26 @@ _PyEval_EvalFrameDefault(PyFrameObject *f, int throwflag)
                           assert(STACK_LEVEL() <= co->co_stacksize); }
 #define POP()           ((void)(lltrace && prtrace(TOP(), "pop")), \
                          BASIC_POP())
-#define STACKADJ(n)     { (void)(BASIC_STACKADJ(n), \
-                          lltrace && prtrace(TOP(), "stackadj")); \
-                          assert(STACK_LEVEL() <= co->co_stacksize); }
+#define STACKADJ_GROW(n)    do { \
+                              assert(n >= 0); \
+                              (void)(BASIC_STACKADJ(n), \
+                              lltrace && prtrace(TOP(), "stackadj")); \
+                              assert(STACK_LEVEL() <= co->co_stacksize); \
+                            } while (0)
+#define STACKADJ_SHRINK(n)  do { \
+                              assert(n >= 0); \
+                              (void)(lltrace && prtrace(TOP(), "stackadj")); \
```

Comment:
This syntax is strange. Why not using a regular if?

```
if (lltrace) prtrace(TOP(), "stackadj");
```

Reply:
I did it like that in order to maintain the the same style as the original author.

I did a quick diff and generate pretty much the same assembly with `-O0 -g` (lltrace is available only on debug builds): https://godbolt.org/g/MMnyrq

Let me know if you prefer a plain if and I'll quickly change